core/src/main/java/org/mskcc/cbio/portal/util/WholeSlideViewerTokenGenerator.java

```diff
@@ -0,0 +1,33 @@
+
+package org.mskcc.cbio.portal.util;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import org.apache.commons.codec.binary.Base64;
+
+public class WholeSlideViewerTokenGenerator {
+	// ~ Static fields
+	// ========================================================================================================
+
+	public static final String PORTAL_NAME = "cbioportal";
+
+	// ~ Methods
+	// ========================================================================================================
+
+	public static String generateToken(String username, String secretKey) {
```

Comment:
i would give this function a more descriptive name

Reply:
is generateTokenByHmacSHA256 better?